src/components/AlarmsMap.tsx

```diff
@@ -9,11 +9,10 @@ import { useRectContext } from "../providers/RectProvider";
 import { WarningArea } from "../types/config";
 import WarningAreaPolygon from "./WarningAreaPolygon";
 import MapCircle from "./MapCircle";
+import { HoverAndSelectContext } from "../providers/HoverAndSelectProvider";
```

Comment:
Two tabs (area alarms and dam alarms) now both need to track which thing is hovered over, or which thing has been clicked to show messages. And the messages thing has precedence, and we also need to be able to show the name of the clicked thing elsewhere, not just its id. This got complicated enough to put into its own context that is used twice.